homeassistant/components/p1_monitor/sensor.py

```diff
@@ -0,0 +1,278 @@
+            device_class=DEVICE_CLASS_VOLTAGE,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="current_phase_l1",
+            name="Current Phase L1",
+            native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+            device_class=DEVICE_CLASS_CURRENT,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="current_phase_l2",
+            name="Current Phase L2",
+            native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+            device_class=DEVICE_CLASS_CURRENT,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="current_phase_l3",
+            name="Current Phase L3",
+            native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+            device_class=DEVICE_CLASS_CURRENT,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_consumed_phase_l1",
+            name="Power Consumed Phase L1",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_consumed_phase_l2",
+            name="Power Consumed Phase L2",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_consumed_phase_l3",
+            name="Power Consumed Phase L3",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_produced_phase_l1",
+            name="Power Produced Phase L1",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_produced_phase_l2",
+            name="Power Produced Phase L2",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+        SensorEntityDescription(
+            key="power_produced_phase_l3",
+            name="Power Produced Phase L3",
+            native_unit_of_measurement=POWER_WATT,
+            device_class=DEVICE_CLASS_POWER,
+            state_class=STATE_CLASS_MEASUREMENT,
+        ),
+    ),
+    SERVICE_SETTINGS: (
+        SensorEntityDescription(
+            key="gas_consumption_tariff",
+            name="Gas Consumption - Tariff",
+            icon="mdi:cash",
+            entity_registry_enabled_default=False,
+            native_unit_of_measurement=CURRENCY_EURO,
```

Comment:
Should the tariff sensors have device class monetary?

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes

Reply:
It should indeed be used